core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -351,17 +349,15 @@ protected static void parseInnerToXContent(XContentParser parser, Builder contex
                 context.setSeqNo(parser.longValue());
             } else if (_PRIMARY_TERM.equals(currentFieldName)) {
                 context.setPrimaryTerm(parser.longValue());
-            } else {
-                throwUnknownField(currentFieldName, parser.getTokenLocation());
             }
         } else if (token == XContentParser.Token.START_OBJECT) {
             if (_SHARDS.equals(currentFieldName)) {
                 context.setShardInfo(ShardInfo.fromXContent(parser));
             } else {
-                throwUnknownField(currentFieldName, parser.getTokenLocation());
+                parser.skipChildren(); // skip potential inner objects for forward compatibility
             }
         } else {
-            throwUnknownToken(token, parser.getTokenLocation());
+            parser.skipChildren(); // skip potential inner arrays for forward compatibility
```

Comment:
nit: shall we be more accurate in verifying that the current token is a start_array? anyways skipChildren won't have any effect otherwise.